### ISSUE TITLE
User interface improvements (member browser).

### ISSUE THREAD

**Issue by meteos77** [textbrowser/biblioteq#348]
After some feedback from volunteers, we've come back to several of them. 

In the member's browser, when editing, when saving, no sign from BiblioteQ that the modifications felt by the user have been taken into account.

Could you please improve this confirmation:
A change of cursor as for borrowing books (a green color) or  a sound like for successful scans.

Thank you in advance

**Comment by textbrowser** [comment]
The positive response is the lack of a negative one.

**Comment by textbrowser** [comment]
If a color is presented, it's going to be forgotten. And lots of colors are lots of confusion. So, no color without text. And text is yet another column that needs to be present for a brief period. BQ is not a purchase system. You make a purchase, you see a page confirming it. See GitHub. You post a message, you see the results. You don't have a color or a popup.

**Comment by textbrowser** [comment]
BQ produces errors if it needs to. It cannot produce a color or a popup or a message for every positive result. It's impractical.

**Comment by textbrowser** [comment]
Web sites have tools for confirmation. Desktops are way different. And because they're way different, the tools are not necessarily portable. Sounds are incomplete because they don't work everywhere. Colors, again, require that BQ remembers lots of states because I guarantee people will forget their purpose. "Why is this green cell green? What did I do? What does it mean?" I'm not adding positive popups and I'm slowly moving away from negative ones too. There's an Error Log. Anyway, I may spend one or two minutes looking.

**Comment by meteos77** [comment]
I suspected your answer before I asked :-)

I'm just passing on the perception of my "colleagues".

For the moment I can't figure out on another program how it handles this action.

Generally, when you press a button, another window opens, a window closes, there's something visual that's released.

**Comment by meteos77** [comment]
Compared with libreoffice, the background color of the button returns less quickly to its basic state

**Comment by meteos77** [comment]
https://github.com/textbrowser/biblioteq/assets/56431039/6deb7162-021a-4359-ad43-77e21fcd1353


https://github.com/textbrowser/biblioteq/assets/56431039/64be2cdf-8d81-4e8c-a6fc-2f0ea741f06e

**Comment by meteos77** [comment]
[LibreOffice.webm](https://github.com/textbrowser/biblioteq/assets/56431039/b55ca3d4-cc2a-4387-91c5-f5f745c3599f)

**Comment by textbrowser** [comment]
Apply is not Apply and Close. Positive feedback is an art. Existing everywhere will dilute it. You're connected to a wireless device. The positivity is a symbol or data flow. Not a popup confirming that you're connected. You press an Apply, the expectation is that something worked. If it did not, there's something else to look at. A log elsewhere. For a life-support system, this sort of thinking is documented, analyzed. It is a core part of the system.

BQ is not life support. You kind of have to trust it. And it includes errors when it fails.

In addition, it does not have insight into PQ errors which may be caused because your wireless connection died. To do that, it requires itself to become aware. I did not for another project and it's intense.

Too many expectations.

**Comment by textbrowser** [comment]
I believe every Apply in BQ results in an error log and / or a warning message. That was a lot of work. I will not enhance that will artificial positive confirmations because they cannot be standardized.

**Comment by meteos77** [comment]
No problem in BiblioteQ, merci pour les explications
it was a request from the volunteers (they are elderly ... their vision is probably less able to detect the change in color of the button, like their reflexes).

could this be due to the choice of colors for the desktop?

**Comment by textbrowser** [comment]
Large blue or green or red buttons indicating a surreal experience. Press and your confusion will vanish. This requires magic because the logic isn't free. People have expectations that are logical because their perceptions urge them to believe in simplicity when the actuality of the matter are far more complex.

**Comment by textbrowser** [comment]
A problem of ideas is that they require completion. Say, as this example illustrates, a widget is activated resulting in a failure. Say, we highlight the widget using an animation while the person who activated the widget is distracted. There must be a separate response that is present until it is acknowledged or discarded. This response must be apparent. It cannot be brief, it cannot be hidden, it cannot be silent.

If this is a heart- or oxygen-monitoring system, the response is simple. An auditory repeating signal. A blaring red or orange signature. A vibrating monitor. A strobe light. There are guidelines which provide suggestions on universal alerts.

BQ is not a place for such a system because it's actually very expensive to provide correct, persistent, uniform alerts. An activated widget which requires the main thread to momentarily pause while an action is completed cannot provide a natural response. In vital systems, there are separate tasks for monitoring, separate tasks for alerting, separate tasks for computations.

TL;DR;

BQ reports failures.

**Comment by textbrowser** [comment]
I be added a floating tool tip to the pictured panel. Float away in blue happiness or red misery. Float away home in color.

**Comment by meteos77** [comment]
Thank you for taking into account the difficulties of my fellow volunteers, I'll get back to you.
For me, it's great!

**Comment by textbrowser** [comment]
Will try to animate the button instead of a floating tool tip.

**Comment by textbrowser** [comment]
This is so much work for an animated button. I need to create a special class which inherits QPushButton. Then, an instance of this class is attached to the UI file. Then, QPropertyAnimation will somehow animate the background color of the push button. However, I don't see it being able to animate a specific property so this is the reason for the specialized QPushButton. I do not see a clear method for animating a widget.

**Comment by textbrowser** [comment]
How this?

https://github.com/textbrowser/biblioteq/assets/10701156/7b600824-ff41-442e-aabc-304d4e50e6a8

**Comment by textbrowser** [comment]
Cannot guarantee that the original color of the button is presented after the animation completes. It looks like it does but there are many styles and distributions and distribution styles.

**Comment by meteos77** [comment]
Super it's a good animation !

**Comment by meteos77** [comment]
in the release note see the line : 
<li>Reokaced QString(tr()) with tr(). Incomplete.</li>
<li>Replaced QString(tr()) with tr(). Incomplete.</li>

**Comment by textbrowser** [comment]
I see. I type with a stylus in one hand.

**Comment by textbrowser** [comment]
Because of KDE Connect. Try it. Transform a tablet into a virtual mouse-less mouse. Prevents painful hands. Ergonomic mice are terrible. With a tablet and a style, one can levitate to happiness.

**Comment by textbrowser** [comment]
Does you place have manga?

**Comment by textbrowser** [comment]
Or books with images and words?

**Comment by meteos77** [comment]
The manga have all been withdrawn because they haven't been released in 5 years!

We have Adult comics and Youth comics. First reading albums for 3 year olds ... 4 years old ...

**Comment by textbrowser** [comment]
He writes about life and being miserable as kids and being bullied. And fantastical monsters. Everything with him is an allegory. Well he illustrates for other reasons too, but there are many characterizations of life concepts. I suppose there's depth and character present. He's not a Marie Curie or Camus or Mann or Sylvia Plath or whatever writer you prefer, but he's like a himself. More adult or late teens.

**Comment by textbrowser** [comment]
An animated review.

https://filmobsessive.com/film/film-analysis/tekkonkinkreet-is-a-forgotten-landmark-in-animation/

The review isn't animated, it's a review of the animated interpretation of the illustrated work.

**Comment by textbrowser** [comment]
Marie Curie still radiates radiation figuratively and literally. She was awesome.

**Comment by textbrowser** [comment]
> The manga have all been withdrawn because they haven't been released in 5 years!
> 
> We have Adult comics and Youth comics. First reading albums for 3 year olds ... 4 years old ...

Withdrawn from the library? Released as in checked out by people?

**Comment by meteos77** [comment]
Donated

**Comment by textbrowser** [comment]
And you don't have purchasing funds. I would purchase a Mac Mini for your project there but I'm not rich. You could leave it active and it'd help the MacOS perception of BQ. Some other person desires PQ on Mac and that's much work because Apple is unlike Microsoft; their products have short lives.

**Comment by textbrowser** [comment]
So donate them to strangers. Like donate them as forbidden content. Label the materials as contraband and everyone will want them. And then you can charge small fees.

**Comment by textbrowser** [comment]
Set up a show in an alleyway at night and have a bunch of volunteers in trench coats offering bootleg books.

**Comment by textbrowser** [comment]
And ask the local po-po to drive by and stage an arrest. Would spike interest in your little library. Macron would visit.

**Comment by textbrowser** [comment]
Funds will pour in and a new building will be erected. And all your money woes will be solved.

**Comment by meteos77** [comment]
We are helped once a year by the town hall to buy 30 - 40 books.
After that, our collection is replenished by our central library, which exchanges 500-600 books with us every 6 months.
We can order books on loan every month.
For example, I've seen your comics on the Louvre (volume 1 and volume 2) ordered, and they'll arrive on Wednesday 6th :-)

**Comment by textbrowser** [comment]
Book printers in the US generally are crap. Mass production and poor quality. Toilet paper is nicer.

**Comment by textbrowser** [comment]
Comics are different. The pages are thicker, the printing finer, the binding is sometimes sewn, and there is a sense of love imbued in the particle. Anyway, good luck. See you.

**Comment by meteos77** [comment]
We have very good publishers ... expensive but good

Bonne continuation!

**Comment by meteos77** [comment]
I can't get the animations on the buttons?
Do I need at least QT6?
QT5 + other librairy?

Can you please give me some more information on how to implement this?

**Comment by textbrowser** [comment]
There's only a positive animation on the pictured button. No, another library is not necessary. It should function everywhere.

**Comment by meteos77** [comment]
I can finally enjoy the excitement of the buttons!

With the `qt5ct` tool
style => gtk2 -> doesn't work
other style -> OK, I'm happy!

Thank you for this new feature !!

**Comment by textbrowser** [comment]
Breeze and Fusion are correct.

**Comment by meteos77** [comment]
With my distribution LinuxMint (21.3), the Breeze style is not installed by default.
I replaced Breeze by Fusion in the startup file biblioteq.sh and it works perfectly

Thanks for the modification!
